src/Service.Host/client/src/components/StockPools.js

```diff
@@ -0,0 +1,306 @@
+                    <DialogTitle>Search</DialogTitle>
+                    <DialogContent>
+                        <Search
+                            autoFocus
+                            propertyName={`${c.field}-searchTerm`}
+                            label="defaultLocation"
+                            resultsInModal
+                            resultLimit={100}
+                            value={searchTerm}
+                            loading={storageLocationsSearchLoading}
+                            handleValueChange={(_, newVal) => setSearchTerm(newVal)}
+                            search={searchStorageLocations}
+                            searchResults={storageLocationsSearchResults}
+                            priorityFunction="closestMatchesFirst"
+                            onResultSelect={handleSearchResultSelect}
+                            clearSearch={clearStorageLocation}
+                        />
+                    </DialogContent>
+                    <DialogActions>
+                        <Button onClick={handleClose}>Close</Button>
+                    </DialogActions>
+                </Dialog>
+            </div>
+        );
+    };
+
+    const stockPoolColumns = [
+        {
+            field: 'stockPoolCode',
+            headerName: 'Code',
+            editable: params => params.row?.creating === true,
+            width: 100
+        },
+        { field: 'stockPoolDescription', editable: true, headerName: 'Description', width: 225 },
+        {
+            field: 'dateInvalid',
+            headerName: 'Date Invalid',
+            editable: true,
+            width: 175,
+            valueFormatter: value => value && moment(value).format('DD-MMM-YYYY')
+        },
+        { field: 'stockCategory', editable: true, headerName: 'Stock Category', width: 130 },
+        {
+            field: 'accountingCompanyCode',
+            editable: true,
+            headerName: 'Accounting Company',
+            width: 175,
+            defaultFormat: true,
+            valueOptions: accountingCompany?.map(ac => ac.name),
+            type: 'singleSelect'
+        },
+        {
+            field: 'defaultLocationName',
+            headerName: 'Default Location',
+            width: 150,
+            editable: true,
+            type: 'search',
+            searchResults: storageLocationsSearchResults,
+            searchLoading: storageLocationsSearchLoading,
+            searchUpdateFieldNames: [
+                { fieldName: 'defaultLocation', searchResultFieldName: 'locationId' },
+                { fieldName: 'defaultLocationName', searchResultFieldName: 'locationCode' }
+            ],
+            clearSearch: clearStorageLocation,
+            renderCell: searchRenderCell
+        },
+        { field: 'availableToMrp', editable: true, headerName: 'Available to Mrp', width: 175 },
+        { field: 'sequence', editable: true, headerName: 'Sequence', width: 100 }
```

Comment:
I was reading over this code again to remind myself how to render a Search inside a DataGrid and it occurred to me:

Why are you passing all these search related props in the DataGrids column definition? Can't you just pass them directly to the Search component? I think passing them via the columns definition only to have to fish them out again is  adding needless complexity